src/Engine/ProtoCore/Lang/Replication/ReplicationInstruction.cs

```diff
@@ -44,5 +45,22 @@ public override string ToString()
             }
 
         }
+
+        public Boolean Equals(ReplicationInstruction oldOption) {
+
+            if (this.Zipped == oldOption.Zipped)
+            {
+                if (this.ZipIndecies == null && oldOption.ZipIndecies == null)
+                    return true;
```

Comment:
why is this true even if the zip algorithm is different?

Reply:
Add the zipAlgorithm check.